suite/tests/api/detach_state.c

```diff
@@ -568,12 +627,74 @@ START_FILE
         pop      r12 @N@ \
         pop      REG_XBX @N@ \
         pop      REG_XBP @N@
-#else /* X64 */
+#elif defined(X86) && !defined(X64)
+/**************************************************
+ * X86_32
+ */
 #    define PUSHALL \
         pusha
 #    define POPALL  \
         popa
 #    error NYI /* TODO i#3160: Add 32-bit support. */
+#elif defined(AARCH64)
+/**************************************************
+ * AARCH64
+ */
+#    define PUSH_GPRS                           \
+        /* SP is checked separately so value here doesn't matter. */ \
+        stp      x30, x0, [sp, #-16]! @N@\
+        stp      x28, x29, [sp, #-16]! @N@\
+        stp      x26, x27, [sp, #-16]! @N@\
+        stp      x24, x25, [sp, #-16]! @N@\
+        stp      x22, x23, [sp, #-16]! @N@\
+        stp      x20, x21, [sp, #-16]! @N@\
+        stp      x18, x19, [sp, #-16]! @N@\
+        stp      x16, x17, [sp, #-16]! @N@\
+        stp      x14, x15, [sp, #-16]! @N@\
+        stp      x12, x13, [sp, #-16]! @N@\
+        stp      x10, x11, [sp, #-16]! @N@\
+        stp      x8, x9, [sp, #-16]! @N@\
+        stp      x6, x7, [sp, #-16]! @N@\
+        stp      x4, x5, [sp, #-16]! @N@\
+        stp      x2, x3, [sp, #-16]! @N@\
+        stp      x0, x1, [sp, #-16]!
+#        define PUSHALL                         \
+        PUSH_GPRS
+        /* TODO i#4698: Push and check SIMD regs. */
+#        define PUSH_CALLEE_SAVED \
+        stp      x28, x29, [sp, #-16]! @N@\
+        stp      x26, x27, [sp, #-16]! @N@\
+        stp      x24, x25, [sp, #-16]! @N@\
+        stp      x22, x23, [sp, #-16]! @N@\
+        stp      x20, x21, [sp, #-16]! @N@\
+        stp      x19, x30, [sp, #-16]!
```

Comment:
Curious why x30 is pushed at the end, paired with x19, which seems a bit irregular. Why not push (x29, x30),(x27,x28),...,(x19,x20)? 

We need ` <even,even+1>` only in ARM right?

Reply:
I think this was b/c I looked up the callee-saved regs and it said x19..x29.  I then copied the lines from PUSH_GPRS and had x19 left over for a non-pair push.  But later I realized x30 as the link register has to be saved too and so I changed the x19 to a pair push.  I will go ahead and rewrite it to make it in order.